src/Majora/Framework/Model/LazyPropertiesTrait.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Majora\Framework\Model;
+
+/**
+ * Trait which provides lazy load delegates call on fields.
+ */
+trait LazyPropertiesTrait
+{
+    /**
+     * @var EntityCollection
+     */
+    private $delegatesCollection;
+
+    /**
+     * @see LazyPropertiesInterface::registerLoader()
+     */
+    public function registerLoader($field, \Closure $delegate)
```

Comment:
I think using `\Closure` is too restricting for users, just `callable` would do IMHO